java-checks/src/main/java/org/sonar/java/checks/RedundantThrowsDeclarationCheck.java

```diff
@@ -55,19 +57,26 @@
 @RspecKey("S1130")
 public class RedundantThrowsDeclarationCheck extends IssuableSubscriptionVisitor {
 
+  private static final Pattern THROWS_JAVADOC_PATTERN = Pattern.compile(".*(?:@throws|@exception)\\s++(?<clazz>\\S*)(\\s++)?(?<descr>.+)?");
+
   @Override
   public List<Tree.Kind> nodesToVisit() {
     return ImmutableList.of(Tree.Kind.METHOD, Tree.Kind.CONSTRUCTOR);
   }
 
   @Override
   public void visitNode(Tree tree) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. ([squid:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1974&issues=AWKPucFeCiQRpXGouqtT&open=AWKPucFeCiQRpXGouqtT)

[ISSUE_KEY]: AWKPucFeCiQRpXGouqtT